backend/main-service/src/main/java/kkakka/mainservice/cart/ui/CartController.java

```diff
@@ -2,28 +2,55 @@
 
 import kkakka.mainservice.cart.application.CartService;
 import kkakka.mainservice.cart.ui.dto.CartRequestDto;
-import org.springframework.beans.factory.annotation.Autowired;
+import kkakka.mainservice.cart.ui.dto.CartResponseDto;
+import kkakka.mainservice.member.domain.Member;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
 
 @RestController
-@RequestMapping("/cart")
+@RequestMapping("/carts")
 public class CartController {
 
-    CartService cartService;
+    private final CartService cartService;
 
-    @Autowired
     public CartController(CartService cartService) {
         this.cartService = cartService;
     }
 
-    @PostMapping("/carts")
+    /* 장바구니 추가 */
+    @PostMapping("/cart")
     public ResponseEntity<String> saveCartItem(@RequestBody CartRequestDto cartRequestDto) {
-        cartService.saveCartItem(cartRequestDto);
 
-        return ResponseEntity.status(HttpStatus.CREATED).body("success");
+        Long memberId = cartService.saveOrUpdateCartItem(cartRequestDto);
+        URI location = ServletUriComponentsBuilder.fromCurrentRequest()
+                .path("/{memberId}")
+                .buildAndExpand(memberId)
+                .toUri();
+
+        return ResponseEntity.created(location).build();
```

Comment:
```suggestion
        Long memberId = cartService.saveOrUpdateCartItem(cartRequestDto);
        return ResponseEntity.created(URI.create(memberId.toString())).build();
```
```suggestion
        Long memberId = cartService.saveOrUpdateCartItem(cartRequestDto);
        return ResponseEntity.created(URI.create(String.format("/%d", memberId))).build();
```

이런식으로 static 메서드를 활용해도 될 것 같습니다~

Reply:
ServletUriComponentsBuilder.fromCurrentRequest() 가 없어서 저대로 하면 Location: /1 이 반환되는데 .... 어떻게 해야할까요

Reply:
다시 보니 ServletUriComponentsBuilder 그대로 쓰는거 좋습니다!👍